src/app/Console/Commands/RequireDevTools.php

```diff
@@ -135,27 +141,93 @@ public function handle()
+            if ($type !== Process::ERR && $buffer !== '') {
+                $details = json_decode($buffer);
+            } elseif (File::exists('auth.json')) {
+                $details = json_decode(File::get('auth.json'), true)['http-basic']['backpackforlaravel.com'] ?? false;
+            }
             $this->progressBar->advance();
         });
 
-        // Finish
-        $this->progressBar->finish();
-        $this->info(' DevTools is now required.');
+        // Create an auth.json file
+        if (! $details) {
+            $this->info(' Creating auth.json file with your authentication token');
 
-        // DevTools inside installer
-        $this->info('');
-        $this->info(' Now running the DevTools installation command.');
+            $this->line(' (Find your access token details on https://backpackforlaravel.com/user/tokens)');
+            $username = $this->ask('Access token username');
+            $password = $this->ask('Access token password');
 
-        // manually include the command in the run-time
-        if (! class_exists(\Backpack\DevTools\Console\Commands\InstallDevTools::class)) {
-            include base_path('vendor/backpack/devtools/src/Console/Commands/InstallDevTools.php');
+            $process = new Process(['composer', 'config', 'http-basic.backpackforlaravel.com', $username, $password]);
+            $process->run(function ($type, $buffer) use ($username, $password) {
+                if ($type === Process::ERR) {
+                    // Fallback
+                    $authFile = [
+                        'http-basic' => [
+                            'backpackforlaravel.com' => [
+                                'username' => $username,
+                                'password' => $password,
+                            ],
+                        ],
+                    ];
+
+                    if (File::exists('auth.json')) {
+                        $currentFile = json_decode(File::get('auth.json'), true);
+                        if (! ($currentFile['http-basic']['backpackforlaravel.com'] ?? false)) {
+                            $authFile = array_merge_recursive($authFile, $currentFile);
+                        }
+                    }
+
+                    File::put('auth.json', json_encode($authFile, JSON_PRETTY_PRINT));
+                }
+
+                $this->progressBar->advance();
+            });
         }
+    }
 
-        $this->call(\Backpack\DevTools\Console\Commands\InstallDevTools::class);
+    public function requireDevTools()
+    {
+        // Require package
+        $process = new Process(['composer', 'require', '--dev', '--with-all-dependencies', 'backpack/devtools']);
+        $process->setTimeout(300);
+        $process->run(function ($type, $buffer) use ($process) {
+            if ($type === Process::ERR) {
+                // create a log channel
+                $log = new Logger('devtools_errors');
```

Comment:
Hmm... why are we using the Monolog\Logger here and not Laravel's `Log` facade? In order to log to a separate file? In that case... wouldn't this force us to add monolog as a dependency?

Also on this... logging might be useful if WE want to debug this. But not for the end-user. What would be useful for the end-user would be to _show what went wrong_. In the command line. Right?

Reply:
Hey @tabacitu . Monolog is a Laravel dependency. 
https://github.com/laravel/framework/blob/8febec76411872306b58aeb1d9f4224e8c1c2ad1/composer.json#L28

We already show a `devtools was not installed correctly` message:

![image](https://user-images.githubusercontent.com/7188159/139878071-bd8cfdd5-5d74-45c1-850e-0b3b99790af3.png)

Laravel provides a way of writting to custom log files https://laravel.com/docs/8.x/logging#on-demand-channels but I am not using the `Log` facade because it won't work out of the box with "any" configuration, so to ensure that we don't rely on user application configuration I just skip one step ahead of it and configured it myself for the desired purpose.

Maybe we should point user to this log file for more information, or show the full `$buffer` to the user in the console, if that is the case, we can probably just skip this log file, or it could be usefull for build commands? Maybe it's still worth it.

Best,
Pedro

Reply:
Ok, I understand. In that case let's:
- [ ] write to the normal log file; I don't think there's much benefit in having this extra file;
- [ ] change the error message - instead of "_please try again_" let's do "_please follow the manual installation process outlined documented in https://backpackforlaravel.com/products/devtools_"

That should:
- make DevTools behave as all other Laravel tools/packages (never heard of one creating its own log file... but we could consider doing this in the future if we find any benefit to it);
- tell the user to repeat things in a way that would at least show different results - the second time, if they run the installation process manually, they'll see the composer conflict error;

Right?

Reply:
@tabacitu I did the changes 👌
It's now writing to the default file, and the message is now better;

```text
┌────────────────────────────────┐
│   DevTools was not installed   │
└────────────────────────────────┘
│ For further information please check the log file.
│ You can also follow the manual installation process documented in https://backpackforlaravel.com/products/devtools
```